src/js/classes/resizable.js

```diff
@@ -0,0 +1,503 @@
+        if ( this._position.left + this._size.width >= this._parentData.width ) {
+          this._size.width = this._parentData.width - this._position.left;
+
+          h = this._size.height;
+          if ( this._ratio ) {
+            this._size.height = this._size.width / this._ratio;
+            continueResize = false;
+          }
+
+          if (this._axis === 'ne' || this._axis === 'n') {
+            this._position.top += h - this._size.height; 
+          }
+        }
+
+        // Too big height case.
+        if ( this._position.top + this._size.height >= this._parentData.height ) {
+          this._size.height = this._parentData.height - this._position.top;
+
+          w = this._size.width;
+          if ( this._ratio ) {
+            this._size.width = this._size.height * this._ratio;
+            continueResize = false;
+          }
+
+          if (this._axis === 'sw') {
+            this._position.left += w - this._size.width;
+          }
+        }
+
+        if ( !continueResize ) {
+          data.left = this._position.left;
+          data.top = this._position.top;
+          data.width = this._size.width;
+          data.height = this._size.height;
+        }
+
+        return data;
+      };
+
+
+      /**
+       * Getters / setters.
+       */
+      ResizableFactory.prototype.setOriginalSize = function(element) {
+        this._originalSize = {
+          width: parseInt(element.css('width'), 10) || element[0].clientWidth,
+          height: parseInt(element.css('height'), 10) || element[0].clientHeight
+        };
+
+        // Updating original size should update _size value.
+        this._size = angular.copy(this._originalSize);
+
+        return this;
+      };
+
+      ResizableFactory.prototype.getOriginalSize = function() {
+        return this._originalSize;
+      };
+
+      ResizableFactory.prototype.setParentSize = function(parentElement) {
+        this._parentData = { 
+          width: parseInt(parentElement.css('width'), 10) || parentElement[0].clientWidth,
+          height: parseInt(parentElement.css('height'), 10) || parentElement[0].clientHeight
+        };
+
+        return this;
+      };
+
+      ResizableFactory.prototype.getParentSize = function() {
+        return this._parentData;
+      };
+
+      ResizableFactory.prototype.setOriginalPosition = function(top, left) {
+        this._originalPosition = { top: top, left: left };
+
+        // Updating original position should update _position value.
+        this._position = angular.copy(this._originalPosition);
+
+        return this;
+      };
+
+      ResizableFactory.prototype.getOriginalPosition = function() {
+        return this._originalPosition;
+      };
+
+      ResizableFactory.prototype.setOriginalMousePosition = function(top, left) {
+        this._originalMousePosition = { top: top, left: left };
+
+        return this;
+      };
+
+      ResizableFactory.prototype.getOriginalMousePosition = function() {

```

Comment:
Can you add a return type to this method? 
Just to know if it returns an int, or an object and if is an object can you tell what type of object sor an example in the method signature?

A good example to take ideas: https://github.com/airbnb/javascript#comments